### PR TITLE
Fix error handling with NME

### PR DESCRIPTION
**Pull request by deltakosh**

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Review by sebavan — APPROVED** [review]

**Comment by Popov72** [comment]
Shouldn't we also do this in the node geometry editor (and me in the render graph editor)?

**Comment by deltakosh** [comment]
Agreed! I'll cover nge to provide the same p

File: packages/dev/core/src/Materials/Node/nodeMaterialBuildStateSharedData.ts [hunk 1]
```diff
@@ -2,8 +2,9 @@ import type { NodeMaterialConnectionPoint } from "./nodeMaterialBlockConnectionP
 import type { NodeMaterialBlock } from "./nodeMaterialBlock";
 import type { InputBlock } from "./Blocks/Input/inputBlock";
 import type { Scene } from "../../scene";
-import type { Immutable } from "../../types";
+import type { Immutable, Nullable } from "../../types";
 import type { NodeMaterial, NodeMaterialTextureBlocks } from "./nodeMaterial";
+import { Logger, type Observable } from "core/Misc";
```
Comment (RaananW) [inline packages/dev/core/src/Materials/Node/nodeMaterialBuildStateSharedData.ts:7]:
```suggestion
import { Logger } from "core/Misc/logger";
import type { Observable } from "core/Misc/observable";
```

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/15524/merge

Test environment: 
https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15524/merge/index.html

To test a playground add it to the URL, for example:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15524/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/15524/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/15524/merge
https://gui.babylonjs.com/?snapshot=refs/pull/15524/merge
https://nme.babylonjs.com/?snapshot=refs/pull/15524/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/15524/merge#BCU1XR#0

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15524/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15524/merge/testResults/webgpuplaywright/index.html

**Pull request merged**